src/CompanyService.Broker/Consumers/ChangeUserPositionConsumer.cs

```diff
@@ -14,8 +14,8 @@ public class ChangeUserPositionConsumer : IConsumer<IChangeUserPositionRequest>
 
         private object ChangePosition(IChangeUserPositionRequest request)
         {
-            _repository.Remove(request.UserId);
             bool isSuccess = _repository.Add(_mapper.Map(request.PositionId, request.UserId));
+            _repository.Remove(request.UserId);
```

Comment:
for what? If you don't wanna delete position before add, then add condition for this. Now you anyway go to Remove. 
You can do following thing: if (!isSuccess) { _repository.Remove(request.UserId); }

Reply:
reworked